identify-layer-features/src/main/java/com/esri/arcgismaps/sample/identifylayerfeatures/components/MapViewModel.kt

```diff
@@ -0,0 +1,145 @@
+        mapViewState.value.arcGISMap = map
+    }
+
+    /**
+     * Identify the feature layer results and display the resulting information
+     */
+    fun handleIdentifyResult(result: Result<List<IdentifyLayerResult>>) {
+        sampleCoroutineScope.launch {
+            result.onSuccess { identifyResultList ->
+                bottomTextBanner.value = ""
+                val message = StringBuilder()
+                var totalCount = 0
+                identifyResultList.forEach { identifyLayerResult ->
+                    val geoElementsCount = geoElementsCountFromResult(identifyLayerResult)
+                    val layerName = identifyLayerResult.layerContent.name
+                    message.append(layerName).append(": ").append(geoElementsCount)
+
+                    // add new line character if not the final element in array
+                    if (identifyLayerResult != identifyResultList[identifyResultList.size - 1]) {
+                        message.append("\n")
+                    }
+                    totalCount += geoElementsCount
+                }
+                // if any elements were found show the results, else notify user that no elements were found
+                if (totalCount > 0) {
+                    bottomTextBanner.value = "Number of elements found:\n${message}"
+                } else {
+                    messageDialogVM.showMessageDialog(
+                        title = "No element found",
+                        description = message.toString()
+                    )
+                }
+            }.onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    title = "Error identifying results: ${error.message.toString()}",
+                    description = error.cause.toString()
+                )
+            }
+        }
+    }
+
+    /**
+     * Gets a count of the GeoElements in the passed result layer.
+     * This method recursively calls itself to descend into sublayers and count their results.
+     * @param result from a single layer.
+     * @return the total count of GeoElements.
+     */
+    private fun geoElementsCountFromResult(result: IdentifyLayerResult): Int {
+        var subLayerGeoElementCount = 0
+        for (sublayerResult in result.sublayerResults) {
+            // recursively call this function to accumulate elements from all sublayers
+            subLayerGeoElementCount += geoElementsCountFromResult(sublayerResult)
+        }
+        return subLayerGeoElementCount + result.geoElements.size
+    }
+}
+
+/**
+ * Data class that represents the MapView state
+ */
+data class MapViewState( // This would change based on each sample implementation
+    var arcGISMap: ArcGISMap = ArcGISMap(BasemapStyle.ArcGISNavigationNight),
+    var viewpoint: Viewpoint = Viewpoint(
+        Point(
+            -10977012.785807, 4514257.550369,
+            SpatialReference(wkid = 3857)
+        ), 68015210.0
+    )
```

Comment:
Couple nested constructors being accessed here, makes it somewhat confusing on how the viewpoint is being set. Maybe we could format it using named arguments to help better clarify the parameters: 

```suggestion
    var viewpoint: Viewpoint = Viewpoint(
        center = Point(
            x = -10977012.785807,
            y = 4514257.550369,
            spatialReference = SpatialReference(wkid = 3857)
        ),
        scale = 68015210.0
    )
```